src/components/Dashboard/Tiles/Widgets/DraggableList.tsx

```diff
@@ -0,0 +1,119 @@
+import { Collapse, createStyles, Stack, Text } from '@mantine/core';
+import { IconChevronDown, IconGripVertical } from '@tabler/icons';
+import { Reorder, useDragControls } from 'framer-motion';
+import { FC, ReactNode, useState } from 'react';
+import { IDraggableListInputValue } from '../../../../widgets/widgets';
+
+const useStyles = createStyles((theme) => ({
+  container: {
+    display: 'flex',
+    flexDirection: 'column',
+    borderRadius: theme.radius.md,
+    border: `1px solid ${
+      theme.colorScheme === 'dark' ? theme.colors.dark[5] : theme.colors.gray[2]
+    }`,
+    backgroundColor: theme.colorScheme === 'dark' ? theme.colors.dark[5] : theme.white,
+    marginBottom: theme.spacing.xs,
+    gap: theme.spacing.xs,
+  },
+  row: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    padding: '12px 16px',
+    gap: theme.spacing.sm,
+  },
+  middle: {
+    flexGrow: 1,
+  },
+  symbol: {
+    fontSize: 16,
+  },
+  clickableIcons: {
+    color: theme.colorScheme === 'dark' ? theme.colors.dark[1] : theme.colors.gray[6],
+    cursor: 'pointer',
+    userSelect: 'none',
+    transition: 'transform .3s ease-in-out',
+  },
+  rotate: {
+    transform: 'rotate(180deg)',
+  },
+  collapseContent: {
+    padding: '12px 16px',
+  },
+}));
+
+type DraggableListParams = {
+  value: IDraggableListInputValue['defaultValue'];
+  onChange: (value: IDraggableListInputValue['defaultValue']) => void;
+  labels: Record<string, string>;
+  children?: Record<string, ReactNode>;
+};
+
+export const DraggableList: FC<DraggableListParams> = (props) => {
+  const keys = props.value.map((v) => v.key);
+
+  return (
+    <div>
+      <Reorder.Group
+        axis="y"
+        values={keys}
+        onReorder={(order) =>
+          props.onChange(order.map((key) => props.value.find((v) => v.key === key)!))
+        }
+        as="div"
+      >
+        {props.value.map((item) => (
+          <ListItem key={item.key} item={item} label={props.labels[item.key]}>
+            {props.children?.[item.key]}
+          </ListItem>
+        ))}
+      </Reorder.Group>
+    </div>
+  );
+};
+
+const ListItem: FC<{
+  item: IDraggableListInputValue['defaultValue'][number];
+  label: string;
+}> = (props) => {
+  const { classes, cx } = useStyles();
+  const controls = useDragControls();
+
+  const [showContent, setShowContent] = useState(false);
```

Comment:
If you want, you could use https://mantine.dev/hooks/use-disclosure/. Then you could simply use the toggle method of that to toggle the collapsable.

Reply:
I found that ``useDisclosure`` sometimes didn't work reliably for me. ``open`` always worked, but ``close`` never did. Am I doing something wrong?

Reply:
Just tried it with `useDisclosure` and it works great! Using `toggle`, as I only have a toggle-like button anyways.